File: day01/python/part1.py
#!/usr/bin/env python3

import helper
from itertools import combinations

GOAL = 2020

def main():
    # fname = "example.txt"
    fname = "input.txt"
    numbers = helper.read_lines_as_ints(fname)

    for a, b in combinations(numbers, 2):
        if a + b == GOAL:
            print(f"# {a}, {b}")
            print()
            result = a * b
            print(result)
            break

##############################################################################

if __name__ == "__main__":
    main()

File: day01/cs/Part1.cs
using static System.Console;
using System.Linq;
using System.Collections.Generic;
using static AdventOfCode2020.Itertools;

namespace AdventOfCode2020
{
    public class Part1
    {
        public static void Main(string[] args)
        {
            WriteLine("# Part 1");
            WriteLine("--------");
            // const string fname = "example.txt";
            const string fname = "input.txt";
            const int GOAL = 2020;
            var numbers = FileUtils.ReadLinesAsInts(fname);
            // WriteLine(string.Join(", ", numbers));
            WriteLine();

            foreach (IEnumerable<int> iter in Combinations(numbers, 2))
            {
                var a = iter.ElementAt(0);
                var b = iter.ElementAt(1);

                if (a + b == GOAL)
                {
                    WriteLine($"# {a}, {b}");
                    WriteLine();
                    var result = a * b;
                    WriteLine($"result: {result}");

                    break;
                }
            }
        }
    }
}
